### PR TITLE
Refactor/ornl-pip-ci

### PR DESCRIPTION
**Pull request by RamonAra209**
- Created a ornl ci configuration for pip on `Summit`
- Modified SDK_WORK_DIR to reflect ORNL [standard](https://github.com/ExaWorks/SDK/tree/master/.gitlab#environment-variables)
- Forced usage of python 3.8.10

### CODE REVIEW

File: .gitlab/ornl-ci-pip.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  stage: build
+  script:
+    - source ${PIP_WORK_DIR}/${VENV_ENV_NAME}/bin/activate
+    - pip install --no-cache-dir -r ${PIP_REQUIREMENTS}
+    - *finalize
+
+.pip_test:
+  stage: test
+  variables:
+    MAINTAINER: "arambula2@llnl.gov"
```
Comment (mtitov) [inline .gitlab/ornl-ci-pip.yml:53]: @danlaney Dan, whom should we assign as maintainer here?

File: .gitlab/ornl-ci-pip.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    IM_NUMBER: "LLNL-MI-834241"
```
Comment (mtitov) [inline .gitlab/ornl-ci-pip.yml:54]: this can be removed for ORNL

File: .gitlab/ornl-ci-pip.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      done
+    - python3 ci/tests/test.py -e
+    - deactivate
+
+### JOBS
+
+# SUMMIT
+pip_env_setup_summit:
+  variables:
+    RUNNER_TYPE: "shell"
```
[the remark this answers is not in the document]
Comment (RamonAra209) [reply .gitlab/ornl-ci-pip.yml:71]: Done

File: .gitlab/ornl-ci-pip.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  extends: [ .on_summit, .job_tags, .pip_env_setup ]
+
+pip_build_summit:
+  variables:
+    RUNNER_TYPE: "shell"
```
[the remark this answers is not in the document]
Comment (RamonAra209) [reply .gitlab/ornl-ci-pip.yml:76]: Done

Also, in job `pip_cleanup`, the runner type was shell, so I switched to `nobatch` as well

File: .gitlab/ornl-ci-pip.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+.on_summit:
+  variables:
+    SITE: "summit"
+    VENV_ENV_NAME: "summit-env"
+
+
+### SCRIPTS
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks ${PIP_WORK_DIR} || true
```
[the remark this answers is not in the document]
Comment (RamonAra209) [reply .gitlab/ornl-ci-pip.yml:23]: Done

File: .gitlab/ornl-ci-pip.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
+variables:
+  PIP_WORK_DIR: ${SDK_WORK_DIR_OPEN}/pip
```
[the remark this answers is not in the document]
Comment (RamonAra209) [reply .gitlab/ornl-ci-pip.yml:2]: Done

File: .gitlab/ornl-ci-pip.yml (hunk 2 above)
Comment (RamonAra209) [reply .gitlab/ornl-ci-pip.yml]: Done

File: .gitlab/ornl-ci-pip.yml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+pip_build_summit:
+  variables:
+    RUNNER_TYPE: "shell"
+  needs: [ pip_env_setup_summit ]
+  extends: [ .on_summit, .job_tags, .pip_build ]
+
+pip_test_summit:
+  variables:
+    RUNNER_TYPE: "batch"
+    LLNL_SLURM_SCHEDULER_PARAMETERS: "--nodes=1 -t 45"
```
[the remark this answers is not in the document]
Comment (mtitov) [reply .gitlab/ornl-ci-pip.yml]: sorry, I made a typo, should be `SCHEDULER_PARAMETERS: "-P CSC449 -nnodes 1 -W 45"`

[the remark this answers is not in the document]
Comment (RamonAra209) [reply .gitlab/ornl-ci-pip.yml]: Done

**Comment by mtitov** [comment]
@RamonAra209 we don't have Summit runners yet (will take some time to figure out how to configure that), let's switch from Summit to Ascent:
- remove Summit jobs, but keep `.on_summit`
- add Ascent machine
- rework `PIP_WORK_DIR` handling - remove it from global variables, and keep it per machine definition

```
.on_ascent:
  variables:
    SITE: "ascent"
    VENV_ENV_NAME: "ascent-env"
    PIP_WORK_DIR: ${SDK_WORK_DIR_OPEN}/pip

.on_summit:
  variables:
    SITE: "summit"
    VENV_ENV_NAME: "summit-env"
    PIP_WORK_DIR: ${SDK_WORK_DIR_ALPINE}/pip
```

and we will have a cleanup per machine here, since all these machines use different FSs
p.s. also I've added a line from Misha's PR https://github.com/ExaWorks/SDK/pull/208 (recreate corresponding directory every 5 days)

```
.pip_cleanup:
  stage: cleanup
  when: always
  script:
   - find ${PIP_WORK_DIR} -maxdepth 0 -type d -ctime +5 | xargs rm -rf
   - *finalize

.....

pip_cleanup_ascent:
  variables:
    RUNNER_TYPE: "nobatch"
  extends: [ .on_ascent, .job_tags, .pip_cleanup ]
```

**Review by mtitov — APPROVED** [review]
LGTM! Good work! Thanks!

**Pull request merged**